azext_iot/iothub/providers/state.py

```diff
@@ -213,6 +266,120 @@ def save_state(self, filename: str, overwrite_file=False):
         except FileNotFoundError:
             raise FileOperationError(f'File {filename} does not exist.')
 
+    @contextmanager
+    def capture_stderr(self):
```

Comment:
After making the adjust to an in-memory buffer, I think this function is generically useful and should be moved to common.utility